chemie/elections/models.py

```diff
@@ -24,244 +25,316 @@
+        user.profile.save()
+        return
+
+    def vote_blank(self, user):
+        blank_ticket = Ticket.create_blank_ticket()
+        self.tickets.add(blank_ticket)
+        user.profile.voted = True
+        user.profile.save()
+        return
 
 
 class Election(models.Model):
     # For the entire election
     is_open = models.BooleanField(verbose_name="Er åpent", default=False)
     positions = models.ManyToManyField(
-        Position, blank=True, related_name="election"
+        Position,
+        blank=True,
+        related_name="election_position",
+        verbose_name="Verv",
     )
     current_position = models.ForeignKey(
         Position,
         blank=True,
         null=True,
         related_name="current_election",
         on_delete=models.CASCADE,
+        verbose_name="Vervet som skal stemmes på",
     )
-    # For sub-elections
-    current_position_is_open = models.BooleanField(
-        verbose_name="Det er åpent for stemming", default=False
-    )
+
     date = models.DateField(auto_now_add=True, blank=True)
 
     def __str__(self):
         return "{}: {}".format(self.id, self.date)
 
-    def add_position(self, positions):
-        if type(positions) is Position:
-            self.positions.add(positions)
-            self.save()
-            return
-        elif type(positions) is QuerySet:
-            self.positions.add(positions)
-            self.save()
-            return
-        elif type(positions) is list:
-            self.positions.add(*positions)
-            self.save()
-        else:
-            raise AttributeError
+    @classmethod
+    def current_position_is_active(cls):
+        if cls.latest_election_is_open():
+            election = cls.get_latest_election()
+            if election.current_position is not None:
+                if election.current_position.is_active:
+                    return True
+        return False
 
-    def delete_position(self, positions):
-        if type(positions) is Position:
-            position = positions
-            position.delete_candidates(candidates=position.candidates.all())
-            position.delete()
-            self.save()
-            return
-        elif type(positions) is QuerySet:
-            deletables = positions
-        elif type(positions) is list:
-            ids = [position.id for position in positions]
-            deletables = Position.objects.filter(pk__in=ids)
+    @classmethod
+    def latest_election_is_open(cls):
+        "Checking if the latest election object is currently open"
+        try:
+            election = cls.objects.latest("id")
+            if election.is_open:
+                return True
+        except ObjectDoesNotExist:  # TODO: more exceptions?
+            pass
+        return False
+
+    @classmethod
+    def create_new_election(cls):
+        Election.objects.create(is_open=True)
+        cls.clear_all_checkins()  # setter alle RFID-checkins til False (ingen har møtt opp enda)
```

Comment:
Kan tenkes at det er lurt å skrive i koden / instrukser for administrering av valg at valget MÅ lages på forhånd, ellers må alle scanne kortet sitt på nytt.

Reply:
Kanskje heller lage en instrukt på admin index siden?

Reply:
Ja, kan skrive en instruks-side hvor vi peker til denne for admin før han åpner valg + at vi har en link til instruksene under oppretting av posisjoner og alt det der når valget først er åpent. Så link til instrukser både før og etter valget et åpnet.